dll/win32/browseui/CAutoComplete.cpp

```diff
@@ -272,17 +272,23 @@ EditWordBreakProcW(LPWSTR lpch, INT index, INT count, INT code)
         {
             if (index)
                 --index;
-            while (index && !IsWordBreak(lpch[index]))
+            while (index &&
+                (!IsWordBreak(lpch[index - 1]) ||(IsWordBreak(lpch[index]) && IsWordBreak(lpch[index - 1]))))
+            {
                 --index;
+            }
             return index;
         }
         case WB_RIGHT:
         {
             if (!count)
                 break;
-            while (index < count && lpch[index] && !IsWordBreak(lpch[index]))
+            while (index < count && lpch[index + 1] &&
```

Comment:
This looks overly complicated. As you've written this, it increments the index if:
* Index < count and:
* The next character after the current one exists and:
* The current character and the next character are not word breaks.
Then it returns `index + 1`.

This appears to be a more complicated version of what already exists which increments the index if:
* Index < count and:
* The current character exists and:
* The current character is not a word break
Then it returns `index`.